ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/state/beaconstate/common/ValidatorIndexCache.java

```diff
@@ -62,31 +59,18 @@ public Optional<Integer> getValidatorIndex(
     return findIndexFromState(state.getValidators(), publicKey);
   }
 
-  private Optional<Integer> findIndexFromState(
-      final SszList<Validator> validatorList, final BLSPublicKey publicKey) {
-    for (int i = Math.max(lastCachedIndex.get() + 1, 0); i < validatorList.size(); i++) {
-      BLSPublicKey pubKey = validatorList.get(i).getPublicKey();
-      // from electra, only finalized validator indices are stable,
-      // so we look to store only finalized indices not already stored.
-      validatorIndices.invalidateWithNewValue(pubKey, i);
-      updateLastIndex(i);
```

Comment:
this is basically a bug... 
basically the net effect here is that the second lookup would fail because it'll scan after that index and miss cache if you're in a NOOP scenario (test added)
Normally its not a noop, and would work, so not awful in deneb currently. fixing now and adding a test stops the problem though.